gammapy/spectrum/dataset.py

```diff
@@ -0,0 +1,167 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import numpy as np
+from .observation import SpectrumObservation
+from .utils import SpectrumEvaluator
+from ..utils.fitting import Dataset, Parameters
+from ..stats import wstat
+
+__all__ = ["SpectrumDatasetOnOff"]
+
+
+class SpectrumDatasetOnOff(Dataset):
+    """Compute spectral model fit statistic on a ON OFF Spectrum.
```

Comment:
Well the point of the `Dataset` framework is to provide the likelihood for a given reduced dataset and a model.
 See the introduction of PIG 8:
'A dataset bundles the reduced data with a parameteric model and fit statistics function. It evaluates the model and log-likelihood and passes it on to the fit object. Datasets can be combined by adding their log-likelihood values and concatenating their model parameters.'

Maybe the term 'fit statistic' is a bit mis-leading, we preferred it to log-likelihood for instance, because it is never really a log-likelihood but rather Cash or wstat statistics.